src/line.rs

```diff
@@ -1619,4 +1633,235 @@ mod tests {
+
+        let initial_registers = new_registers();
+        let opcode = Opcode::Special(16);
+        let expected_registers = initial_registers.clone();
+
+        assert_exec_opcode(header, initial_registers, opcode, expected_registers, true);
+    }
+
+    #[test]
+    fn test_exec_special_negative_line_advance() {
+        let header = make_test_header(&[]);
+
+        let mut initial_registers = new_registers();
+        initial_registers.line = 10;
+
+        let opcode = Opcode::Special(13);
+
+        let mut expected_registers = initial_registers.clone();
+        expected_registers.line -= 3;
+
+        assert_exec_opcode(header, initial_registers, opcode, expected_registers, true);
+    }
+
+    #[test]
+    fn test_exec_special_positive_line_advance() {
+        let header = make_test_header(&[]);
+
+        let initial_registers = new_registers();
+
+        let opcode = Opcode::Special(19);
+
+        let mut expected_registers = initial_registers.clone();
+        expected_registers.line += 3;
+
+        assert_exec_opcode(header, initial_registers, opcode, expected_registers, true);
+    }
+
+    #[test]
+    fn test_exec_special_positive_address_advance() {
+        let header = make_test_header(&[]);
+
+        let initial_registers = new_registers();
+
+        let opcode = Opcode::Special(52);
+
+        let mut expected_registers = initial_registers.clone();
+        expected_registers.address += 3;
+
+        assert_exec_opcode(header, initial_registers, opcode, expected_registers, true);
+    }
+
+    #[test]
+    fn test_exec_special_positive_address_and_line_advance() {
+        let header = make_test_header(&[]);
+
+        let initial_registers = new_registers();
+
+        let opcode = Opcode::Special(55);
+
+        let mut expected_registers = initial_registers.clone();
+        expected_registers.address += 3;
+        expected_registers.line += 3;
+
+        assert_exec_opcode(header, initial_registers, opcode, expected_registers, true);
+    }
+
+    #[test]
+    fn test_exec_special_positive_address_and_negative_line_advance() {
+        let header = make_test_header(&[]);
+
+        let mut initial_registers = new_registers();
+        initial_registers.line = 10;
+
+        let opcode = Opcode::Special(49);
+
+        let mut expected_registers = initial_registers.clone();
+        expected_registers.address += 3;
+        expected_registers.line -= 3;
+
+        assert_exec_opcode(header, initial_registers, opcode, expected_registers, true);
+    }
+
+    #[test]
+    fn test_exec_special_line_underflow() {
+        let header = make_test_header(&[]);
+
+        let mut initial_registers = new_registers();
+        initial_registers.line = 2;
+
+        // -3 line advance.
+        let opcode = Opcode::Special(13);
+
+        let mut expected_registers = initial_registers.clone();
+        // Clamp at 0. No idea if this is the best way to handle this situation
+        // or not...
```

Comment:
I expect most other DWARF implementations will wrap. I don't really care what we do until we encounter a case that needs some specific behavior to work.